### ISSUE TITLE
HunyuanVideo lora trained with Musubi tuner doesn't work in framepack

### ISSUE THREAD

**Issue by danielkorg** [kohya-ss/musubi-tuner#293]
Here are the commands I used to train a HunyuanVideo lora with this framework:

python cache_latents.py --dataset_config dataset.toml --vae pytorch_model.pt --vae_chunk_size 32 --vae_tiling

python cache_text_encoder_outputs.py --dataset_config dataset.toml --text_encoder1 llava_llama3_fp16.safetensors --text_encoder2 clip_l.safetensors --batch_size 1 --fp8_llm

then, I tried both with and without the accelerate launch

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 
python hv_train_network.py --dit mp_rank_00_model_states.pt --dataset_config dataset.toml --split_attn --flash_attn --blocks_to_swap=20 --mixed_precision bf16 --fp8_base --optimizer_type came_pytorch.CAME --learning_rate 2e-4 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module networks.lora --network_dim 16 --network_alpha 1 --gradient_accumulation_steps 1 --network_args "loraplus_lr_ratio=4" --timestep_sampling shift --discrete_flow_shift 7.0 --lr_warmup_steps 100 --lr_scheduler constant_with_warmup --max_train_steps 2400 --save_every_n_steps 400 --seed 42 --output_dir ./output --output_name framepacklora

The readme says to configure accelerate for single gpu with these settings, so i did:

- In which compute environment are you running?: This machine
- Which type of machine are you using?: No distributed training
- Do you want to run your training on CPU only (even if a GPU / Apple Silicon / Ascend NPU device is available)?[yes/NO]: NO
- Do you wish to optimize your script with torch dynamo?[yes/NO]: NO
- Do you want to use DeepSpeed? [yes/NO]: NO
- What GPU(s) (by id) should be used for training on this machine as a comma-seperated list? [all]: all
- Would you like to enable numa efficiency? (Currently only supported on NVIDIA hardware). [yes/NO]: NO
- Do you wish to use mixed precision?: bf16

dataset.toml contents (dataset is like 25 pics):

[general]
resolution = [512, 768]
caption_extension = ".txt"
batch_size = 1
enable_bucket = true
bucket_no_upscale = false

[[datasets]]
image_directory = "framepackstuff"
num_repeats = 1

Then, when I use this lora inside a framepack repo like: https://github.com/colinurbs/FramePack-Studio
it gives the error: Error loading LoRA framepacklora: chunk expects at least a 1-dimensional tensor

i know i can use the lora with the fpack_generate_video, but i dont care about that, i wanted to train a
hunyuan video lora with this and then use it in any framepack GUI framework like the one i mentioned (studio).

i also did some digging online and that error seems to be linked to a problem with https://huggingface.co/docs/safetensors/torch_shared_tensors
but im not 100% sure...

anyway, long story short, would be awesome to make it work somehow with any framepack framework that has
lora loaders builtin. Is there a way to maybe convert the output lora trained to a format that would be able to
load inside framepack? is there any hint as to how to make it work? https://github.com/colinurbs/FramePack-Studio/blob/f8327fff7c45a3be66644c641a051b865cba2d0d/diffusers_helper/lora_utils.py

any help is appreciated!

**Comment by danielkorg** [comment]
I did a test with the --save_merged_model argument and it does technically work but this is not ideal at all..

I wish I knew what the difference was in the framepack-studio lora loader and your loader that makes it work
on yours and not the studio version...

**Comment by danielkorg** [comment]
ah no problem! would be awesome <3 thank you so much for the hard work!

**Comment by maybleMyers** [comment]
I have made a gui around musubi's inference scripts, with some additions to support extra inference features, that you might find helpful. https://github.com/maybleMyers/H1111

**Comment by arledesma** [comment]
> I think that LoRA of HunyuanVideo of Musubi Tuner does not work with FramePack-Studio because it is a different format from diffusion-pipe. Also, LoRA of FramePack based on FramePack model, not HunyuanVideo, so that seems to be another problem.
> 
> I am thinking of opening a PR on FramePack-Studio for the convenience of the users. However, in order to do so, I need to understand FramePack-Studio, which may take some time.

@kohya-ss Your LoRAReady implementation has been adapted into FPS. It's currently baking in the develop branch.
https://github.com/FP-Studio/framepack-studio/tree/3ca1f549f3219c139bf5ff639b14ff502e99c8d7/diffusers_helper/lora_utils_kohya_ss

**Comment by kohya-ss** [comment]
> [@kohya-ss](https://github.com/kohya-ss) Your LoRAReady implementation has been adapted into FPS. It's currently baking in the develop branch.
> https://github.com/FP-Studio/framepack-studio/tree/3ca1f549f3219c139bf5ff639b14ff502e99c8d7/diffusers_helper/lora_utils_kohya_ss

That's great! and I'm sure it will help a lot of users. I will close this issue once that branch is merged.
